TrackletAlgorithm/TrackHandler.cc

```diff
@@ -0,0 +1,86 @@
+#include "TrackHandler.h"
+
+bool TrackHandler::compareTrack(TrackStruct& trk, TrackStruct& masterTrk, unsigned int& matchFound, unsigned int mergeCondition){
+  #pragma HLS inline
+  // compare the two tracks, masterTrack and trk
+  LOOP_CompareTrack:
+  for (unsigned int layerIndex = 0; layerIndex < 4; layerIndex++){
+    #pragma HLS unroll
+    auto masterBarrelStubIndex = ap_uint<TrackFit<NBarrelStub, NDiskStub>::kTFStubIndexSize>(masterTrk._barrelStubArray[layerIndex][0].range(kBarrelStubIndexSizeMSB, kBarrelStubIndexSizeLSB));
+    auto inputBarrelStubIndex = ap_uint<TrackFit<NBarrelStub, NDiskStub>::kTFStubIndexSize>(trk._barrelStubArray[layerIndex][0].range(kBarrelStubIndexSizeMSB, kBarrelStubIndexSizeLSB));
+    auto masterDiskStubIndex = ap_uint<TrackFit<NBarrelStub, NDiskStub>::kTFStubIndexSize>(masterTrk._diskStubArray[layerIndex][0].range(kDiskStubIndexSizeMSB, kDiskStubIndexSizeLSB));
+    auto inputDiskStubIndex = ap_uint<TrackFit<NBarrelStub, NDiskStub>::kTFStubIndexSize>(trk._diskStubArray[layerIndex][0].range(kDiskStubIndexSizeMSB, kDiskStubIndexSizeLSB));
+
+    if((masterBarrelStubIndex == inputBarrelStubIndex) && (masterBarrelStubIndex > 0)){
+      matchesFoundBarrel[layerIndex][0] = 1;
+    } else {
+      matchesFoundBarrel[layerIndex][0] = 0;
+    }
+    
+    if ((masterDiskStubIndex == inputDiskStubIndex) && (masterDiskStubIndex > 0)){
+      matchesFoundDisk[layerIndex][0] = 1;
+    } else {
+      matchesFoundDisk[layerIndex][0] = 0;
+    }
+  }
+
+  // update #matches found 
+  unsigned int matchesFound = 0;
+  LOOP_MatchesFound:
+  for (unsigned int layerIndex = 0; layerIndex < 4; layerIndex++){
+    #pragma HLS unroll
+    matchesFound += matchesFoundBarrel[layerIndex][0] + matchesFoundDisk[layerIndex][0];
+  }
+  if (matchesFound >= mergeCondition){
+    matchFound = 1;
+    return matchFound; 
+    } else {
+      matchFound = 0;
+      return matchFound;
+    }
+}
+
+void TrackHandler::mergeTrack(TrackStruct& trk, TrackStruct& masterTrk){
+  #pragma HLS inline
+  // update master track
+  // check whether the stub word is non-zero in the compared track
+  // then add stub into master track if share > 3 stubs in common
+  LOOP_SetInputStubToMaster:
+  for(unsigned int layerIndex = 0; layerIndex < 4; layerIndex++){
+    #pragma HLS unroll
+    if((masterTrk._barrelStubArray[layerIndex][0] == 0) && (trk._barrelStubArray[layerIndex][0] != 0)){
+      masterTrk._barrelStubArray[layerIndex][0] = trk._barrelStubArray[layerIndex][0];
+      mergedBarrelStubsMap += (1 << 3*layerIndex);
```

Comment:
Please replace the `3` with `TrackFit<NBarrelStub, NDiskStub>::kTFHitCountSize` (or a constant derived from this).